airbyte-cdk/python/airbyte_cdk/sources/declarative/retrievers/simple_retriever.py

```diff
@@ -408,20 +409,39 @@ def state(self, value: StreamState):
     def parse_records_and_emit_request_and_responses(self, request, response, stream_slice, stream_state) -> Iterable[StreamData]:
         # Only emit requests and responses when running in debug mode
         if self.logger.isEnabledFor(logging.DEBUG):
-            yield self._create_trace_message_from_request(request)
-            yield self._create_trace_message_from_response(response)
+            yield prepared_request_to_airbyte_message(request)
+            yield response_to_airbyte_message(response)
         # Not great to need to call _read_pages which is a private method
         # A better approach would be to extract the HTTP client from the HttpStream and call it directly from the HttpRequester
         yield from self.parse_response(response, stream_slice=stream_slice, stream_state=stream_state)
 
-    def _create_trace_message_from_request(self, request: requests.PreparedRequest):
-        # FIXME: this should return some sort of trace message
-        request_dict = {"url": request.url, "http_method": request.method, "headers": dict(request.headers), "body": request.body}
-        log_message = filter_secrets(f"request:{json.dumps(request_dict)}")
-        return AirbyteMessage(type=MessageType.LOG, log=AirbyteLogMessage(level=Level.INFO, message=log_message))
-
-    def _create_trace_message_from_response(self, response: requests.Response):
-        # FIXME: this should return some sort of trace message
-        response_dict = {"body": response.text, "headers": dict(response.headers), "status_code": response.status_code}
-        log_message = filter_secrets(f"response:{json.dumps(response_dict)}")
-        return AirbyteMessage(type=MessageType.LOG, log=AirbyteLogMessage(level=Level.INFO, message=log_message))
+
+def prepared_request_to_airbyte_message(request: requests.PreparedRequest) -> AirbyteMessage:
+    # FIXME: this should return some sort of trace message
+    request_dict = {
+        "url": request.url,
+        "http_method": request.method,
+        "headers": dict(request.headers),
+        "body": _body_binary_string_to_dict(request.body),
+    }
+    log_message = filter_secrets(f"request:{json.dumps(request_dict)}")
+    return AirbyteMessage(type=MessageType.LOG, log=AirbyteLogMessage(level=Level.INFO, message=log_message))
+
+
+def _body_binary_string_to_dict(body_str) -> Optional[Mapping[str, str]]:
+    if body_str:
+        if isinstance(body_str, (bytes, bytearray)):
+            body_str = body_str.decode()
+        try:
+            return json.loads(body_str)
+        except JSONDecodeError:
+            return {k: v for k, v in [s.split("=") for s in body_str.split("&")]}
+    else:
+        return None
+
+
+def response_to_airbyte_message(response: requests.Response) -> AirbyteMessage:
```

Comment:
moved to a function for consistency and ease of testing

Reply:
Those methods aren't used anywhere else except from the tests though. Should we indicate them as private using the `_` prefix? 

Saying that makes me think it is weird that `parse_records_and_emit_request_and_responses` is public as well since it's only used internally by the class. My main concern is that we have someone external using it which would bring a dependency to manage when we want to change this. This happened for https://github.com/airbytehq/airbyte/pull/20019 and prevented us to do a breaking change we wanted to do